src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificateAssetDownloader.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Internal.Cryptography.Pal
+{
+    internal static class CertificateAssetDownloader
+    {
+        private static unsafe Interop.libcurl.curl_unsafe_write_callback s_writeCallback = CurlWriteCallback;
+
+        static CertificateAssetDownloader()
+        {
+            Interop.libcurl.curl_global_init(Interop.libcurl.CurlGlobalFlags.CURL_GLOBAL_ALL);
```

Comment:
While that may be true, at the very least we need to ensure we're not invoking this function concurrently with itself, which we might be doing now if one thread used HttpClient at the same time another is using X509. We also need to ensure that we've properly initialized OpenSSL when using HttpClient. That all implies we need a native initialization shim for libcurl that both libraries can use, just as we do for SSL, and that shim will need to use our SSL initialization routine as well. Right?